### PR TITLE
Added a new helper method for updateJsonDocument

### PR DESCRIPTION
**Pull request by GwilymNewton**
Pull request for Issue #477 , tested against my own code, but I've not extended the test scripts

### CODE REVIEW

**Comment by GwilymNewton** [comment]
Okay, it looks like that travis build has not passed for about 2 months, so let me know if I actually do need to change anything.

**Comment by nfriedly** [comment]
Yea, nobody has been able to put much time into this recently. However, in your case it does need a bit more work: Run `npm run autofix` to handle the lint/style errors and then run `npm test` locally to see if it needs anything else.

File: discovery/v1.js [hunk 1]
```diff
@@ -545,6 +545,32 @@ DiscoveryV1.prototype.addJsonDocument = function(params, callback) {
  * @param {String} params.environment_id environment guid for the collection
  * @param {string} params.collection_id the guid of the collection
  * @param {string} params.document_id the guid of the document to update
+ * @param {string} params.document_id the guid of the document to update
+ * @param {Buffer|ReadableStream|Object} params.file a file to post (smaller than 50mb)
```
Comment (nfriedly) [inline discovery/v1.js]: This line should say 

     * @param {Object} params.file non-stringified JSON object

Comment (GwilymNewton) [inline discovery/v1.js]: @nfriedly Done

**Comment by GwilymNewton** [comment]
@nfriedly 
Here is output of autofix
```
Gwilyms-MacBook-Pro:node-sdk gwilymnewton$ npm run autofix

> watson-developer-cloud@2.34.0 autofix /Users/gwilymnewton/Box Sync/node-sdk
> eslint . --fix


/Users/gwilymnewton/Box Sync/node-sdk/examples/conversation_tone_analyzer_integration/tone_conversation_integration.v1.js
  13:7   error  Unexpected tab character  no-tabs
  15:10  error  Unexpected tab character  no-tabs
  17:7   error  Unexpected tab character  no-tabs
  18:7   error  Unexpected tab character  no-tabs
  19:7   error  Unexpected tab character  no-tabs
  21:7   error  Unexpected tab character  no-tabs
  80:5   error  Unexpected tab character  no-tabs

/Users/gwilymnewton/Box Sync/node-sdk/tradeoff-analytics/v1.js
  48:5  error  Unexpected tab character  no-tabs
  85:5  error  Unexpected tab character  no-tabs

✖ 9 problems (9 errors, 0 warnings)


npm ERR! Darwin 16.6.0
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "run" "autofix"
npm ERR! node v6.10.3
npm ERR! npm  v3.10.10
npm ERR! code ELIFECYCLE
npm ERR! watson-developer-cloud@2.34.0 autofix: `eslint . --fix`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the watson-developer-cloud@2.34.0 autofix script 'eslint . --fix'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the watson-developer-cloud package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     eslint . --fix
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs watson-developer-cloud
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls watson-developer-cloud
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/gwilymnewton/Box Sync/node-sdk/npm-debug.log
```

and test
```
Gwilyms-MacBook-Pro:node-sdk gwilymnewton$ npm test

> watson-developer-cloud@2.34.0 test /Users/gwilymnewton/Box Sync/node-sdk
> npm run lint && mocha test/unit test/integration


> watson-developer-cloud@2.34.0 lint /Users/gwilymnewton/Box Sync/node-sdk
> npm run compat-check && eslint . --cache && dependency-lint


> watson-developer-cloud@2.34.0 compat-check /Users/gwilymnewton/Box Sync/node-sdk
> eslint --print-config .eslintrc.js | eslint-config-prettier-check

No rules that are unnecessary or conflict with prettier were found.

/Users/gwilymnewton/Box Sync/node-sdk/examples/conversation_tone_analyzer_integration/tone_conversation_integration.v1.js
  13:7   error  Unexpected tab character  no-tabs
  15:10  error  Unexpected tab character  no-tabs
  17:7   error  Unexpected tab character  no-tabs
  18:7   error  Unexpected tab character  no-tabs
  19:7   error  Unexpected tab character  no-tabs
  21:7   error  Unexpected tab character  no-tabs
  80:5   error  Unexpected tab character  no-tabs

/Users/gwilymnewton/Box Sync/node-sdk/tradeoff-analytics/v1.js
  48:5  error  Unexpected tab character  no-tabs
  85:5  error  Unexpected tab character  no-tabs

✖ 9 problems (9 errors, 0 warnings)


npm ERR! Darwin 16.6.0
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "run" "lint"
npm ERR! node v6.10.3
npm ERR! npm  v3.10.10
npm ERR! code ELIFECYCLE
npm ERR! watson-developer-cloud@2.34.0 lint: `npm run compat-check && eslint . --cache && dependency-lint`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the watson-developer-cloud@2.34.0 lint script 'npm run compat-check && eslint . --cache && dependency-lint'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the watson-developer-cloud package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     npm run compat-check && eslint . --cache && dependency-lint
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs watson-developer-cloud
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls watson-developer-cloud
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/gwilymnewton/Box Sync/node-sdk/npm-debug.log
npm ERR! Test failed.  See above for more details.
```

File: discovery/v1.js [hunk 2]
```diff
@@ -545,6 +545,31 @@ DiscoveryV1.prototype.addJsonDocument = function(params, callback) {
  * @param {String} params.environment_id environment guid for the collection
  * @param {string} params.collection_id the guid of the collection
  * @param {string} params.document_id the guid of the document to update
+ * @param {Object} params.file non-stringified JSON object
+ * @param {Buffer|ReadableStream|Object} params.file a file to post (smaller than 50mb)
```
Comment (nfriedly) [inline discovery/v1.js]: This still needs to be corrected

Comment (GwilymNewton) [inline discovery/v1.js]: done in d356dfb

Comment (nfriedly) [inline discovery/v1.js]: oh, sorry I didn't even realize there were two definitions

**Comment by nfriedly** [comment]
Thanks, I'll get this released soon

**Comment by GwilymNewton** [comment]
@nfriedly Any chance of getting a release soon?

**Comment by nfriedly** [comment]
Sorry, I've been underwater for the last couple of weeks.  It should be out in `v2.35.0` shortly.

**Pull request merged**